test/e2e/runner/execution.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	e2e "github.com/cometbft/cometbft/test/e2e/pkg"
+	"github.com/cometbft/cometbft/test/e2e/pkg/infra"
+	"github.com/cometbft/cometbft/test/e2e/pkg/infra/docker"
+)
+
+func SaveExecution(testnet *e2e.Testnet) error {
+	logger.Info("saving execution", "msg", "saving e2e network execution information")
+	// Fetch and save the execution logs
+	now := time.Now()
+	timestamp := now.Format("20060102_150405")
+	executionFolder := filepath.Join("networks_executions", testnet.Name, timestamp)
+	if err := os.MkdirAll(executionFolder, 0o755); err != nil {
+		logger.Error("error saving execution", "msg", "error creating executions folder", "err", err.Error())
+		return err
+	}
+	for _, node := range testnet.Nodes {
+		// Pause the container to capture the logs
+		_, err := docker.ExecComposeOutput(context.Background(), testnet.Dir, "pause", node.Name)
+		if err != nil {
+			logger.Error("error saving execution", "msg", "error pausing container", "node", node.Name, "err", err.Error())
+			return err
+		}
+
+		// Get the logs from the Docker container
+		data, err := docker.ExecComposeOutput(context.Background(), testnet.Dir, "logs", node.Name)
+		if err != nil {
+			logger.Error("error saving execution", "msg", "error getting logs from container", "node", node.Name, "err", err.Error())
+			return err
+		}
+
+		// Create a file to write the processed lines
+		nodeFolder := filepath.Join(executionFolder, "nodes", node.Name)
```

Comment:
Do we need this `nodes` sub-directory?

Reply:
maybe not, I was wondering that, maybe if I remove it there might be a possibility you can start `docker-compose` from the execution run directory and since the config and data folders are in there, maybe there's a way to "re-start" the chain for inspection `cometbft inspection` so you could have access to the RPC for retrieving data from the chain (e.g. `/block`)